resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -306,9 +318,14 @@ private[yarn] class YarnAllocator(
       s"executorsStarting: ${numExecutorsStarting.get}")
 
     if (missing > 0) {
-      logInfo(s"Will request $missing executor container(s), each with " +
-        s"${resource.getVirtualCores} core(s) and " +
-        s"${resource.getMemory} MB memory (including $memoryOverhead MB of overhead)")
+      var requestContainerMessage = s"Will request $missing executor container(s), each with " +
+          s"${resource.getVirtualCores} core(s) and " +
+          s"${resource.getMemory} MB memory (including $memoryOverhead MB of overhead) "
+      if (ResourceTypeHelper.isYarnResourceTypesAvailable()) {
+        requestContainerMessage ++= s"and with custom resources:" +
+            ResourceTypeHelper.getCustomResourcesAsStrings(resource)
```

Comment:
I think you should check for the message being an empty string, so you don't confusingly put just `s"and with custom resources:"`

Reply:
As the if condition guarantees that the variable missing is greater than zero and resource.getVirtualCores and resource.getMemory will return valid values in every case, the string can never be an empty string in my opinion.
Please let me know if you have any objections.